KubeArmor/monitor/systemMonitor.go

```diff
@@ -209,6 +239,150 @@ func isIgnored(item string, ignoreList []string) bool {
+	if err != nil {
+		mon.Logger.Warnf("error closing bpf map kubearmor_visibility %v", err)
+	}
+}
+
+// UpdateNsKeyMap Function
+func (mon *SystemMonitor) UpdateNsKeyMap(action string, nsKey NsKey, visibility tp.Visibility) {
+	var err error
+
+	file := cle.MapKV{
+		Key:   uint32(0),
+		Value: visibility_off,
+	}
+	process := cle.MapKV{
+		Key:   uint32(1),
+		Value: visibility_off,
+	}
+	network := cle.MapKV{
+		Key:   uint32(2),
+		Value: visibility_off,
+	}
+	capability := cle.MapKV{
+		Key:   uint32(3),
+		Value: visibility_off,
+	}
+	if visibility.File {
+		file.Value = visibility_on
+	}
+	if visibility.Process {
+		process.Value = visibility_on
+	}
+	if visibility.Capabilities {
+		capability.Value = visibility_on
+	}
+	if visibility.Network {
+		network.Value = visibility_on
+	}
+
+	if action == "ADDED" {
+		spec := mon.BpfVisibilityMapSpec
+		spec.Contents = append(spec.Contents, file)
+		spec.Contents = append(spec.Contents, process)
+		spec.Contents = append(spec.Contents, network)
+		spec.Contents = append(spec.Contents, capability)
+		visibilityMap, err := cle.NewMap(&spec)
+		if err != nil {
+			mon.Logger.Warnf("Cannot create bpf map %s", err)
+			return
+		}
+		mon.NsVisibilityMap[nsKey] = visibilityMap
+		err = mon.BpfNsVisibilityMap.Put(nsKey, visibilityMap)
+		if err != nil {
+			mon.Logger.Warnf("Cannot insert insert visibiliy map into kernel nskey=%+v, error=%s", nsKey, err)
+		}
+		mon.Logger.Printf("Successfully added visibiliy map with key=%+v to the kernel", nsKey)
+	} else if action == "MODIFIED" {
+		visibilityMap := mon.NsVisibilityMap[nsKey]
+		if visibilityMap == nil {
+			mon.Logger.Warnf("Cannot locate visibiliy map. nskey=%+v, action=modified", nsKey)
+			return
+		}
+		err = visibilityMap.Put(file.Key, file.Value)
+		if err != nil {
+			mon.Logger.Warnf("Cannot update visibiliy map. nskey=%+v, value=%+v, scope=file", nsKey)
+		}
+		err = visibilityMap.Put(process.Key, process.Value)
+		if err != nil {
+			mon.Logger.Warnf("Cannot update visibiliy map. nskey=%+v, value=%+v, scope=process", nsKey)
+		}
+		err = visibilityMap.Put(network.Key, network.Value)
+		if err != nil {
+			mon.Logger.Warnf("Cannot update visibiliy map. nskey=%+v, value=%+v, scope=network", nsKey)
+		}
+		err = visibilityMap.Put(capability.Key, capability.Value)
+		if err != nil {
+			mon.Logger.Warnf("Cannot update visibiliy map. nskey=%+v, value=%+v, scope=capability", nsKey)
+		}
+		mon.Logger.Printf("Updated visibiliy map with key=%+v", nsKey)
+	} else if action == "DELETED" {
+		err := mon.BpfNsVisibilityMap.Delete(nsKey)
+		if err != nil {
+			mon.Logger.Warnf("Cannot locate visibiliy map. nskey=%+v, action=deleted", nsKey)
+			return
+		}
+		delete(mon.NsVisibilityMap, nsKey)
+		mon.Logger.Printf("Successfully deleted visibiliy map with key=%+v from the kernel", nsKey)
+	}
+}
+
+// UpdateHostVisibility Function
+func (mon *SystemMonitor) UpdateHostVisibility() {
+	nsKey := NsKey{
+		PidNS: 0,
+		MntNS: 0,
+	}
```

Comment:
Since we already have this code. We should eliminate the need for conditional compilation. and update the in kernel logic to not consider it as well.